node/silkworm/db/access_layer.cpp

```diff
@@ -317,8 +316,8 @@ static std::optional<ByteView> historical_storage(mdbx::txn& txn, const evmc::ad
     }
 
     const ByteView k{from_slice(data.key)};
-    if (k.substr(0, kAddressLength) != full_view(address) ||
-        k.substr(kAddressLength, kHashLength) != full_view(location)) {
+    if (k.substr(0, kAddressLength) != ByteView{address} ||
+        k.substr(kAddressLength, kHashLength) != ByteView{location}) {
```

Comment:
Probably we need to check `data.key.length() >= kAddressLength + kHashLength`.
Should a wrong record be in table then substr would throw an out of bounds exception

Reply:
Right; done in https://github.com/torquem-ch/silkworm/pull/494/commits/ef07eea0b72edaf64b11ac1336b1af8b30b24e9a